galaxy/importer/loaders/base.py

```diff
@@ -46,11 +49,26 @@ def __init__(self, content_type, path, root, logger=None):
 
         self.log = logutils.ContentTypeAdapter(
             logger or default_logger, self.content_type, self.name)
+        self._validate_name()
 
     @property
     def path(self):
         return os.path.join(self.root, self.rel_path)
 
+    def _validate_name(self):
+        """Validate content name if one found from directory path.
+
+        A repository role exists at root, so content name will not exist yet.
+        """
+        if not self.name:
+            return
+        if not re.match(constants.NAME_REGEXP, self.name):
```

Comment:
Not something to address in this pr, but I've found that to raise more specific error messages, I had to split this regex up into smaller parts. ie, for things like the leading numbers, double underscores, etc.

Reply:
Having the error message point users to the right place in documentation regarding the formatting details might be a consideration rather than multiple specific error messages.